src/test/java/net/snowflake/ingest/SimpleIngestIT.java

```diff
@@ -114,61 +138,105 @@ public void testSimpleIngest() throws Exception
 
     assertEquals("SUCCESS", insertResponse.getResponseCode());
 
+    // Get history and ensure that the expected file has been ingested
+    getHistoryAndAssertLoad(manager, TEST_FILE_NAME);
+
+    IngestResponse insertResponseSkippedFiles = manager.ingestFile(myFile, null, true);
+
+    assertEquals("SUCCESS", insertResponseSkippedFiles.getResponseCode());
+    assertEquals(1, insertResponseSkippedFiles.getSkippedFiles().size());
+    assertEquals(TEST_FILE_NAME, insertResponseSkippedFiles.getSkippedFiles().stream().findFirst().get());
+  }
+
+  /**
+   * ingest test example
+   * ingest a simple file and check load history.
+   */
+  @Test
+  public void testSimpleIngestWithPattern() throws Exception
+  {
+    //put
+    TestUtils.executeQuery(
+            "put file://" + testFilePath + " @" + stageWithPatternName
+    );
+
+    TestUtils.executeQuery(
+            "put file://" + testFilePath_2 + " @" + stageWithPatternName
+    );
+
+    //create ingest manager
+    SimpleIngestManager manager = TestUtils.getManager(pipeWithPatternName);
+    Set<String> files = new HashSet<>();
+    files.add(TEST_FILE_NAME);
+    files.add(TEST_FILE_NAME_2);
+
+    //get an insert response after we submit
+    IngestResponse insertResponse = manager.ingestFiles(SimpleIngestManager.wrapFilepaths(files), null);
```

Comment:
Should we also verify if showskippedfiles is not present in the response object? https://snowflakecomputing.atlassian.net/browse/SNOW-75404
(Based on this JIRA discussion)

Reply:
i.e if the response should include two newly added fields?
(if they are null/empty we might as well ignore them for being backward compatible?)

Reply:
Currently they are null if the 2 new properties are not returned from the service. I can add an assert for it. One way of ignoring them (for backward compatible reasons) is by versioning the response object and adding new APIs to return the new versioned object, but that seems too heavy and unnecessary. Is there any other way of ignoring them in Java? 

I have a clarifying question regarding back-compat: This change is introducing new behavior but is not changing any existing behavior, isn't it backward compatible already? Or are we considering addition of new properties to the response as a breaking change?

Reply:
I think we can ignore it using jackson annotation. https://www.baeldung.com/jackson-ignore-null-fields

It wont be a breaking change if we release a new version. But by backward compatibility, I meant if customers adopt this new released version, are they expected to change anything if they dont want to utilize new fields? (If my understanding is correct for B/W compatibility)

Reply:
The ignore null in jackson ignores it during serialization from what I can gather, whereas, this will be during de-serialization. 

I don't think the customers have to change anything if they adopt the new version but don't want any of the new fields as these are additional new fields on the response and are read-only. IMO, this shouldn't break back-compat (The basic test worked without any changes needed in them after I added these 2 new fields).

Reply:
I think we are still fine as this doesnt qualify for "no B/W compatible change".
https://bambielli.com/til/2018-01-12-what-is-a-breaking-change/
But, I still feel we should remove null fields.

Reply:
Ahh true for only serialization. Are we ensuring that the response from GS will not contain null fields?

Reply:
Yup, GS is ensuring only non empty fields are returned
https://github.com/snowflakedb/snowflake/blob/a2d9148b9478b1164f6bde1fdfc01eea5089b723/GlobalServices/src/main/java/com/snowflake/ingest/PushFileResponse.java#L12

Reply:
Sounds good, thanks for confirming!